packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -25,6 +25,7 @@ import { PreferenceSchema, PreferenceSchemaProperties } from '@theia/core/lib/co
 import { ProblemMatcherContribution, ProblemPatternContribution, TaskDefinition } from '@theia/task/lib/common';
 import { FileStat } from '@theia/filesystem/lib/common';
 import { ColorDefinition } from '@theia/core/lib/browser/color-registry';
+import * as path from 'path';
```

Comment:
You cannot use `path` (Node.js) API in `common`.

Reply:
why not? It worked in our solution.

Reply:
> why not? It worked in our solution.

It is against the [code organization](https://github.com/eclipse-theia/theia/wiki/Code-Organization) rules. Note, there are CI errors with the current changes.

Reply:
Replaced the `path` import with custom logic to add the `/` if `pck.packagePath` does not end with one. Based on previous usage `relativePath` should never start with a slash, and `pck.packagePath` does not seem to end with one (based on one observation) but this way we should be safe.

Reply:
I think `toPluginUrl` should be a method on real service, which does the URI building for you. Thoughts, @akosyakov?

Reply:
We cannot since it is used in the plugin host process which does not use DI.